Initial implementation of gsc-ec-converter.

itohro opened the pull request:
Tested the converter against 2 projects that have gs-collections as dependencies.

https://github.com/jClarity/slow-crud-app
https://github.com/sMiyoshi/java-collections-sample

gs-rezaem commented on src/main/java/org/eclipse/collections/tools/Converter.java:
```diff
@@ -0,0 +1,105 @@
+                    .forEach(file -> {
+                        try
+                        {
+                            MutableList<String> allLines = ListAdapter.adapt(Files.readAllLines(file, Charset.defaultCharset()));
```
This is not production quality code, but this is not production code, so it's fine.
Typically, reading an entire file into memory to replace strings in it is not the right approach. Line by line would be the right way.

gs-rezaem commented on src/main/java/org/eclipse/collections/tools/Converter.java:
```diff
@@ -0,0 +1,105 @@
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.MalformedInputException;
+import java.nio.file.Files;
```
This requires JDK 7 to compile/run. Is that ok?

gs-rezaem commented on src/main/java/org/eclipse/collections/tools/Converter.java:
```diff
@@ -0,0 +1,105 @@
+        {
+            Files.walk(Paths.get(path))
+                    .filter(Files::isRegularFile)
+                    .filter(file -> IntInterval.fromTo(0, file.getNameCount() - 1).collect(file::getName).noneSatisfy(s -> s.toString().startsWith(".")))
```
why are we filtering out "." files instead of just looking for ".java" files?

itohro commented:
Thanks for the feedback, I've updated below.

* Improved the file read part to stream line by line and update.
* Added files suffix scope .java, .xml and .gradle. Also updated README.md accordingly.

It should be fine to limit to JDK 7 and upwards. This application is meant to run standalone and it's reasonable to discard JDK 6 and below at this point. We can consider if any request is raised to support JDK 6 and below.

motlin commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+gradle.prefs
+.idea/workspace.xml
+.idea/uiDesigner.xml
+src/test/resources/test
```
Why exclude a directory under src/test/resources?

motlin commented on README.md:
````diff
@@ -0,0 +1,51 @@
+
+GS Collections 7.0 and Eclipse Collections 7.0 are functionally equivalent besides the name space changes below.
+
+* Maven groupId: from ```com.goldmansacks``` to ```org.eclipse.collections```
````
typo in the groupId

motlin commented on README.md:
```diff
@@ -0,0 +1,51 @@
+
+To help users migrating from GS Collections to Eclipse Collections, 
+we created a handy converter to replace the dependencies in your project.
+
```
Not just dependencies - alters source code in place.

motlin commented on build.gradle:
```diff
@@ -0,0 +1,40 @@
+ */
+
+group 'org.eclipse.collections'
+version '7.1.0-SNAPSHOT'
```
Should this use a production release (7.0.0) instead of 7.1.0-SNAPSHOT?

itohro commented:
As part of unit test, test files are copied to ```src/test/resources/test```, main method runs against this dir, and cleaned-up afterwards. Added this folder to .gitignore just to make sure anything under this folder is accidentally committed.

Thanks for the feedback in README.md. I've committed changes with fixup so they're incorporated in 657afde.

motlin commented:
Is it possible to use the JUnit Rule for temporary folders instead? Or
target/
On Thu, Feb 11, 2016 at 8:54 AM Hiroshi Ito <notifications@github.com>
wrote:

> As part of unit test, test files are copied to src/test/resources/test,
> main method runs against this dir, and cleaned-up afterwards. Added this
> folder to .gitignore just to make sure anything under this folder is
> accidentally committed.
>
> Thanks for the feedback in README.md. I've committed changes with fixup so
> they're incorporated in 657afde
> <https://github.com/eclipse/gsc-ec-converter/commit/657afde0f2ee259b0a1e37498d2cb66dc21ca9ec>.
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipse/gsc-ec-converter/pull/1#issuecomment-182874926>
> .
>

itohro commented:
Good suggestion indeed. I've refactored the test to use temp dir.
